src/lib.rs

```diff
@@ -365,12 +365,16 @@ fn process_join(from: &Arc<Session>, room_id: RoomId, user_id: UserId, subscribe
 
     let mut is_master_handle = false;
     if let Some(subscription) = subscribe.as_ref() {
-        let max_room_size = STATE.config.get().unwrap().max_room_size;
-        let room_is_full = switchboard.occupants_of(&room_id).len() >= max_room_size;
+        let config = STATE.config.get().unwrap();
+        let room_is_full = switchboard.occupants_of(&room_id).len() > config.max_room_size;
+        let server_is_full = switchboard.sessions().len() > config.max_ccu;
         is_master_handle = subscription.data; // hack -- assume there is only one "master" data connection per user
         if is_master_handle && room_is_full {
             return Err(From::from("Room is full."))
         }
+        if is_master_handle && server_is_full {
+            return Err(From::from("Server is full."))
```

Comment:
hey, you promised error codes! :)

Reply:
(not serious, please don't add error codes, because it means I have to do more work)